### PR TITLE
feat: Allow for text driven icons and content projection

### LINKED ISSUE #6631: Allow for text icons and labels (content projection)
Icons are configurable so that they can be customised without changing the component implementation. There are however a few limitations to the implementation of icons:
- text based icons are not supported (although this could be achieved with CSS pseudo selectors
- icon labels cannot be projected inside an ICON

Moreover, icons have been implemented in a component, where a directive could be beneficial in a lot of occasions. A directive would avoid another host element for the icon and allow for cleaner DOM. 

With this ticket we're adding those features:
- optional usage of a directive over a component
- text based icons
- content projection

A follow up ticket will come up with a technique to add configurable ICON labels.

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
We're adding a few features to configurable icons:
- (optional) icon directive over a component
- text based icons (next to SVG and font driven icons)
- content projection into ICON component or directive

closes #6631

BREAKING CHANGES
- `IconLoaderService` constructor got a new argument to sanitisation
- `IconLoaderService.getSvgPath()` and `IconLoaderService.useSvg() ` are no longer public as the svg handling is managed inside the icon loader service rather than the icon component. 
- the `icon.component.html` contains a `<i>` element hosting the icon html (SVG)
- the `icon.component.ts` is refactored, `useSvg` is no longer available on the component controller as the SVG added inside the service.

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/misc/icon/icon-loader.service.ts [hunk 1]
```diff
@@ -12,31 +14,59 @@ import {
...
+    protected winRef: WindowRef,
+    protected iconConfig: IconConfig,
+    protected sanitizer: DomSanitizer
+  ) {}
 
   /**
-   * Indicates whether the given icon type is configured to use SVG.
+   * Returns an html fragment which can be added to the DOM in a safe way.
    */
-  useSvg(iconType: ICON_TYPE): boolean {
```
Comment (Zeyber) [inline projects/storefrontlib/src/cms-components/misc/icon/icon-loader.service.ts]: Renaming/Deleting method is a breaking change

File: projects/storefrontlib/src/cms-components/misc/icon/icon.component.ts [hunk 2]
```diff
@@ -1,73 +1,81 @@
...
    */
-  private staticStyleClasses: string;
+  icon: SafeHtml;
 
   constructor(
     protected iconLoader: IconLoaderService,
-    protected elementRef: ElementRef<HTMLElement>
+    protected elementRef: ElementRef<HTMLElement>,
+    protected renderer: Renderer2
   ) {}
```
Comment (Zeyber) [inline projects/storefrontlib/src/cms-components/misc/icon/icon.component.ts]: New constructor argument

File: projects/storefrontlib/src/cms-components/misc/icon/icon-loader.service.ts (hunk 1 above)
Comment (Zeyber) [reply projects/storefrontlib/src/cms-components/misc/icon/icon-loader.service.ts]: Same case with `useSvg` on component

Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-components/misc/icon/icon-loader.service.ts]: updated the (prepared) commit message

File: projects/storefrontlib/src/cms-components/misc/icon/icon.component.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/cms-components/misc/icon/icon.component.ts]: yup, that's fine

**Pull request merged**